scri/flux.py

```diff
@@ -437,12 +429,298 @@ def angular_momentum_flux(h, hdot=None):
+    _, ethbar_h_chiminus_ethbar_hdot = matrix_expectation_value(ethbar_h, functools.partial(p_minus, s=-3), ethbar_hdot)
+    _, eth_h_chiplus_eth_hdot = matrix_expectation_value(eth_h, functools.partial(p_plus, s=-1), eth_hdot)
+    _, eth_h_chiminus_eth_hdot = matrix_expectation_value(eth_h, functools.partial(p_minus, s=-1), eth_hdot)
+    _, h_chiplus_hdot = matrix_expectation_value(h, functools.partial(p_plus, s=-2), hdot)
+    _, h_chiminus_hdot = matrix_expectation_value(h, functools.partial(p_minus, s=-2), hdot)
+    _, eth_hdot_eth_chiplus_h = matrix_expectation_value(eth_hdot, functools.partial(eth_chi_plus, s=-2), h)
+    _, eth_hdot_eth_chiminus_h = matrix_expectation_value(eth_hdot, functools.partial(eth_chi_minus, s=-2), h)
+    _, h_eth_chiplus_eth_hdot = matrix_expectation_value(h, functools.partial(ethbar_chi_plus, s=-2), eth_hdot)
+    _, h_eth_chiminus_eth_hdot = matrix_expectation_value(h, functools.partial(ethbar_chi_minus, s=-2), eth_hdot)
+    _, hdot_chiplus_hdot = matrix_expectation_value(hdot, functools.partial(p_plus, s=-2), hdot)
+    _, hdot_chiminus_hdot = matrix_expectation_value(hdot, functools.partial(p_minus, s=-2), hdot)
+
+    # Terms in the z direction are computed here.
+    _, ethbar_hdot_chiz_ethbar_h = matrix_expectation_value(ethbar_hdot, functools.partial(p_z, s=-3), ethbar_h)
+    _, eth_hdot_chiz_eth_h = matrix_expectation_value(eth_hdot, functools.partial(p_z, s=-1), eth_h)
+    _, hdot_chiz_h = matrix_expectation_value(hdot, functools.partial(p_z, s=-2), h)
+    _, ethbar_h_chiz_ethbar_hdot = matrix_expectation_value(ethbar_h, functools.partial(p_z, s=-3), ethbar_hdot)
+    _, eth_h_chiz_eth_hdot = matrix_expectation_value(eth_h, functools.partial(p_z, s=-1), eth_hdot)
+    _, h_chiz_hdot = matrix_expectation_value(h, functools.partial(p_z, s=-2), hdot)
+    _, eth_hdot_eth_chiz_h = matrix_expectation_value(eth_hdot, functools.partial(eth_chi_z, s=-2), h)
+    _, h_ethbar_chiz_eth_hdot = matrix_expectation_value(h, functools.partial(ethbar_chi_z, s=-2), eth_hdot)
+    _, hdot_chiz_hdot = matrix_expectation_value(hdot, functools.partial(p_z, s=-2), hdot)
+
+    # These terms needs to be added to get the components in a particular direction.
+    # The basis will be changed from (plus, minus, z) to (x, y, z).
+    
+    boost_flux_plus = (1 / 8)* (ethbar_hdot_chiplus_ethbar_h - eth_hdot_chiplus_eth_h + 6 * hdot_chiplus_h + ethbar_h_chiplus_ethbar_hdot - eth_h_chiplus_eth_hdot + 6 * h_chiplus_hdot) - (1 / 2) * np.multiply(h.t, hdot_chiplus_hdot) - (1 / 4) * (eth_hdot_eth_chiplus_h - h_eth_chiplus_eth_hdot)
+    
+    boost_flux_minus = (1 / 8) * (ethbar_hdot_chiminus_ethbar_h - eth_hdot_chiminus_eth_h + 6 * hdot_chiminus_h + ethbar_h_chiminus_ethbar_hdot - eth_h_chiminus_eth_hdot + 6 * h_chiminus_hdot) - (1 / 2) * np.multiply(h.t, hdot_chiminus_hdot) - (1 / 4) * (eth_hdot_eth_chiminus_h - h_eth_chiminus_eth_hdot)
```

Comment:
These lines are quite long, please run the formatter `black` over your code. You can manually go back and tweak lines that look really weird but for the most part we are trying to stick to the Black formatting convention. Let me know if you would like some help with this. :grin: